File: hw/sanchez/u3/hw23Singleton/SingletonPy/tax.py
class Tax:
    _instance = None

    def __init__(self):
        self.percentage = 0.0

    @classmethod
    def get_instance(cls):
        if cls._instance is None:
            cls._instance = cls()
        return cls._instance

    def update_tax_percentage(self, percentage):
        self.percentage = percentage

    def get_percentage(self):
        return self.percentage

    def sales_total(self, sale):
        return sale + (sale * self.percentage / 100)

File: hw/sanchez/u3/hw23Singleton/SingletonJs/Tax.js
class Tax {
    static instance = null;

    constructor() {
        this.percentage = 0.0;
    }

    static getInstance() {
        if (Tax.instance === null) {
            Tax.instance = new Tax();
        }
        return Tax.instance;
    }

    updateTaxPercentage(percentage) {
        this.percentage = percentage;
    }

    getPercentage() {
        return this.percentage;
    }

    salesTotal(sale) {
        return sale + (sale * this.percentage / 100);
    }
}

module.exports = Tax;